go/vt/vtgate/evalengine/func.go

```diff
@@ -595,3 +597,29 @@ func aggregatedType(env *ExpressionEnv, expr []Expr) sqltypes.Type {
 	}
 	return sqltypes.VarChar
 }
+
+type builtinFloor struct{}
+
+func (builtinFloor) call(env *ExpressionEnv, args []EvalResult, result *EvalResult) {
+	if args[0].null() {
+		result.setNull()
+		return
+	}
+
+	args[0].makeFloat()
+	val := strconv.FormatFloat(math.Floor(args[0].float64()), 'f', -1, 64)
+
+	result.setString(val, collations.TypedCollation{
```

Comment:
This is most definitely not the way to implement `FLOOR`, as it can never return a `VARCHAR` value. Have you actually ran the tests locally?